src/components/pages/profile/[name]/tabs/OwnershipTab/sections/RolesSection/components/RoleRow.tsx

```diff
@@ -0,0 +1,169 @@
+    display: flex;
+    flex-flow: row wrap;
+    justify-content: space-between;
+    align-items: center;
+
+    gap: ${theme.space[4]};
+  `,
+)
+
+const RoleTagContainer = styled.div(
+  ({ theme }) => css`
+    display: flex;
+    flex-wrap: wrap;
+    gap: ${theme.space[2]};
+  `,
+)
+
+type Props = {
+  name: string
+  address?: string | null
+  roles: Role[]
+  actions: ReturnType<typeof useRoleActions>['data']
+  isWrapped: boolean
+  isEmancipated: boolean
+}
+
+export const RoleRow = ({ name, address, roles, actions, isWrapped, isEmancipated }: Props) => {
+  const router = useRouterWithHistory()
+  const { t } = useTranslation('common')
+
+  const primary = usePrimary(address!, !address)
+  const networkName = useChainName()
+  const wrapperAddress = useContractAddress('NameWrapper')
+  const registrarAddress = useContractAddress('BaseRegistrarImplementation')
+  const [, copy] = useCopyToClipboard()
+
+  const etherscanAction = useMemo(() => {
+    const primaryName = primary.data?.name
+    if (!primaryName) return null
+    const is2ldEth = checkETH2LDFromName(primaryName)
+    const hasToken = is2ldEth || isWrapped
+    if (!hasToken) return null
+    const hex = isWrapped ? namehash(primaryName) : labelhash(primaryName.split('.')[0])
+    const tokenId = BigNumber.from(hex).toString()
+    const contractAddress = isWrapped ? wrapperAddress : registrarAddress
+    return {
+      label: t('transaction.viewEtherscan', { ns: 'common' }),
+      onClick: () =>
+        window.open(
+          makeEtherscanLink(`${contractAddress}/${tokenId}`, networkName, 'nft'),
+          '_blank',
+        ),
+      icon: <OutlinkSVG />,
+    }
+  }, [primary.data?.name, isWrapped, networkName, wrapperAddress, registrarAddress, t])
+
+  const editRolesAction = actions?.find(({ type, disabled }) => type === 'edit-roles' && !disabled)
+
+  const syncManagerAction = roles.includes('manager')
+    ? actions?.find(({ type, disabled }) => type === 'sync-manager' && !disabled)
+    : null
+
+  const items: DropdownItem[] = [
+    ...(primary.data?.name
+      ? ([
+          {
+            label: t('wallet.viewProfile'),
+            onClick: () => router.push(getDestination(`/profile/${primary.data!.name}`) as string),
+            color: 'text',
+            icon: <UpRightArrowSVG />,
+          },
+          {
+            label: t('name.copy'),
+            onClick: () => copy(primary.data!.name!),
+            color: 'text',
+            icon: <CopySVG />,
+          },
+        ] as DropdownItem[])
+      : []),
+    {
+      label: t('address.viewAddress'),
+      onClick: () => router.push(getDestination(`/address/${address}`) as string),
+      color: 'text',
+      icon: <UpRightArrowSVG />,
+    },
+    {
+      label: t('address.copyAddress'),
+      onClick: () => copy(address!),
+      color: 'text',
+      icon: <CopySVG />,
+    },
+    ...(etherscanAction ? [etherscanAction] : []),
+    ...(editRolesAction ? [editRolesAction] : []),
```

Comment:
I can convert it how I did it in other arrays where I add a filter at the end to get rid of empty results. I do think that is cleaner. Having a null value in the array would also require that we make an update in thorin to ignore null values or it will throw errors.